crypto/rsa_extra/rsa_test.cc

```diff
@@ -648,6 +648,57 @@ TEST(RSATest, OnlyDGiven) {
                          buf_len, key.get()));
 }
 
+TEST(RSATest, Set0Key) {
+  const int hash_nid = NID_sha256;
+  const uint8_t kDummyHash[32] = {0};
+  uint8_t sig[256];
+  unsigned sig_len = sizeof(sig);
+
+  // Reference RSA key that needs to be reset before each case as
+  // RSA_set0_key takes ownership of its parameters
+  bssl::UniquePtr<RSA> rsa;
+
+  // Allocate an empty key to imitate JCA keys via calls to RSA_set0_key
+  bssl::UniquePtr<RSA> jcaKey(RSA_new());
+  ASSERT_TRUE(jcaKey);
+
+  // FULL KEY, BLINDING => OK
+  rsa.reset(RSA_private_key_from_bytes(kKey1, sizeof(kKey1) - 1));
+  ASSERT_TRUE(rsa);
+  jcaKey.reset(RSA_new());
+  ASSERT_TRUE(jcaKey);
+  EXPECT_TRUE(RSA_set0_key(jcaKey.get(), rsa->n, rsa->e, rsa->d));
```

Comment:
After this step you are double-referencing memory. Pointers in both `jcaKey` and `rsa` points to the same memory, and the former will contain dangling pointers after L677.

Reply:
ah, that makes sense. for some reason, I don't hit the segfault locally that the CI is hitting. I wrapped these pointers in `BN_dup`. Still passes locally, hopefully the CI agrees.